### PR TITLE
Vendor docker/docker, fsouza and more #2

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

Vendors in fsouza/docker-client, docker/docker and
a few more related. Of particular note, changes to the TweakCapabilities()
function from docker/docker along with the parse.IDMappingOptions() function
from Buildah. Please pay particular attention to the related changes in
the call from libpod to those functions during the review.

Passes baseline tests.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Not sure how I managed to mess up #2627, but this replaces it after #2625 has now been merged.

File: pkg/spec/spec.go [hunk 1]
```diff
@@ -625,7 +625,7 @@ func setupCapabilities(config *CreateConfig, configSpec *spec.Spec) error {
 	if useNotRoot(config.User) {
 		configSpec.Process.Capabilities.Bounding = caplist
 	}
-	caplist, err = caps.TweakCapabilities(configSpec.Process.Capabilities.Bounding, config.CapAdd, config.CapDrop)
+	caplist, err = caps.TweakCapabilities(configSpec.Process.Capabilities.Bounding, config.CapAdd, config.CapDrop, nil, false)
```
Comment (vrothberg) [inline pkg/spec/spec.go]: LGTM :+1:

Comment (mheon) [reply pkg/spec/spec.go]: I'd like @rhatdan to give this a once-over to be sure it looks OK

**Comment by mheon** [comment]
LGTM, but I'd like to hold until @rhatdan does a once-over on the capabilities changes

File: pkg/spec/spec.go (hunk 1 above)
Comment (TomSweeneyRedHat) [reply pkg/spec/spec.go]: Yes, this is the scariest change in here.

**Comment by rhatdan** [comment]
Bizarre.  I see no use for the new fields that they added to TweakCapabilities.
LGTM
/lgtm

**Pull request merged**